### PR TITLE
Common field names rule

### PR DESCRIPTION
**Pull request by cjhowald**
Adds common field names rule.

### CODE REVIEW

File: src/main/java/de/zalando/zally/rules/CommonFieldNamesRule.java [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+            return true;
+        }
+        String expectedFormat = commonFields.get(propName).getFormat();
+        return expectedFormat == null || expectedFormat.equals(property.getFormat());
+    }
+
+    private final String title = "Use common field names";
+    private final String link = "http://zalando.github.io/restful-api-guidelines/common-data-objects/CommonDataObjects.html" +
+            "#must-use-common-field-names";
+    private final ViolationType violationType = ViolationType.MUST;
```
Comment (netme) [inline src/main/java/de/zalando/zally/rules/CommonFieldNamesRule.java]: Optional: It's good to define constants in the very beginning of this class. Will be easier to maintain later.

File: src/main/java/de/zalando/zally/rules/CommonFieldNamesRule.java [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+        }
+
+        for (Map.Entry<String, Model> definitionEntry : swagger.getDefinitions().entrySet()) {
+            String definition = definitionEntry.getKey();
+            Model model = definitionEntry.getValue();
+            for (Map.Entry<String, Property> propertyEntry: model.getProperties().entrySet()) {
+                String name = propertyEntry.getKey();
+                Property property = propertyEntry.getValue();
+                if (!matchesCommonFieldsType(name, property)) {
+                    String description = createDescription(definition, name, "type",property.getType(), commonFields.get(name).getType());
```
Comment (netme) [inline src/main/java/de/zalando/zally/rules/CommonFieldNamesRule.java]: Optional: Space is missing after "type"

File: src/main/java/de/zalando/zally/rules/CommonFieldNamesRule.java [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+                    violations.add(new Violation(title, description, violationType, link));
+                } else if (!matchesCommonFieldsFormat(name, property)) {
+                    String description = createDescription(definition, name, "format",property.getFormat(), commonFields.get(name).getFormat());
```
Comment (netme) [inline src/main/java/de/zalando/zally/rules/CommonFieldNamesRule.java]: Optional: Space is missing after "format"

File: src/main/java/de/zalando/zally/rules/CommonFieldNamesRule.java [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+                    violations.add(new Violation(title, description, violationType, link));
+                }
+            }
+        }
+
+        return violations;
+    }
+
+    private String createDescription(String definition, String propertyName, String problemField, String problemValue, String expectedValue) {
```
Comment (netme) [inline src/main/java/de/zalando/zally/rules/CommonFieldNamesRule.java]: Nice function!

File: src/main/java/de/zalando/zally/rules/CommonFieldNamesRule.java (hunk 1 above)
Comment (cjhowald) [inline src/main/java/de/zalando/zally/rules/CommonFieldNamesRule.java]: Fixed

File: src/main/java/de/zalando/zally/rules/CommonFieldNamesRule.java (hunk 3 above)
Comment (cjhowald) [inline src/main/java/de/zalando/zally/rules/CommonFieldNamesRule.java]: Fixed

File: src/main/java/de/zalando/zally/rules/CommonFieldNamesRule.java (hunk 4 above)
Comment (cjhowald) [inline src/main/java/de/zalando/zally/rules/CommonFieldNamesRule.java]: Thanks

**Pull request merged**